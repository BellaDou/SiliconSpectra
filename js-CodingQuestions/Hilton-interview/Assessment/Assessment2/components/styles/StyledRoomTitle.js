import styled from "styled-components";

const StyledRoomTitle = styled.div`
  padding-left: 0.1em;
  display: flex;
  flex-direction: row;
  justify-content: start;
  align-items: center;
  ${({ room }) => {
    if (!room.ableToCheck || room.checked) {
      return `
      background-color: #ededed;
    `;
    } else {
      return `background-color: #b8b8b8;`;
    }
  }}
  margin-bottom: 0.2em;
`;
export default StyledRoomTitle;
