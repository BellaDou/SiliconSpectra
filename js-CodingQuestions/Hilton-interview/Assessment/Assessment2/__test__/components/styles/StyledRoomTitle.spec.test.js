import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import StyledRoomTitle from "../../../components/styles/StyledRoomTitle";

const checkedRoom = {
  id: 2,
  ableToCheck: true,
  checked: true,
  adults: 1,
  children: 0
};

test("Should properly render style component", () => {
  const tree = renderer.create(<StyledRoomTitle room={checkedRoom} />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("padding-left", "0.1em");
  expect(tree).toHaveStyleRule("display", "flex");
  expect(tree).toHaveStyleRule("flex-direction", "row");
  expect(tree).toHaveStyleRule("justify-content", "start");
  expect(tree).toHaveStyleRule("align-items", "center");
  expect(tree).toHaveStyleRule("margin-bottom", "0.2em");
  expect(tree).toHaveStyleRule("background-color", "#ededed");
});
const uncheckedRoom = {
  id: 3,
  ableToCheck: true,
  checked: false,
  adults: 1,
  children: 0
};

test("Should properly render style component", () => {
  const tree = renderer
    .create(<StyledRoomTitle room={uncheckedRoom} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("padding-left", "0.1em");
  expect(tree).toHaveStyleRule("display", "flex");
  expect(tree).toHaveStyleRule("flex-direction", "row");
  expect(tree).toHaveStyleRule("justify-content", "start");
  expect(tree).toHaveStyleRule("align-items", "center");
  expect(tree).toHaveStyleRule("margin-bottom", "0.2em");
  expect(tree).toHaveStyleRule("background-color", "#b8b8b8");
});
