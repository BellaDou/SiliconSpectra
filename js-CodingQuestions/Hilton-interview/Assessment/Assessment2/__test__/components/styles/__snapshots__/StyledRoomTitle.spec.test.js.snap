// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should properly render style component 1`] = `
.c0 {
  padding-left: 0.1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ededed;
  margin-bottom: 0.2em;
}

<div
  className="c0"
/>
`;

exports[`Should properly render style component 2`] = `
.c0 {
  padding-left: 0.1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b8b8b8;
  margin-bottom: 0.2em;
}

<div
  className="c0"
/>
`;
